src/VCR/Cassette.php

```diff
@@ -68,17 +68,45 @@ public function hasResponse(Request $request)
      */
     public function playback(Request $request)
     {
-        foreach ($this->storage as $recording) {
-            $storedRequest = Request::fromArray($recording['request']);
-            if ($storedRequest->matches($request, $this->getRequestMatchers())) {
-                return Response::fromArray($recording['response']);
+        if ($this->config->getPlaybackMode() === VCR::PLAYBACK_MODE_ALL_IN_ORDER) {
+            if ($this->storage->valid()) {
+                $recording = $this->storage->current();
+                $this->storage->next();
+
```

Comment:
When the cassette is loaded in memory it will keep track of it's current position. In this step we get the recording at the current step (both request and response) and move the current position to next available. When then attempt to match the recording with the current request. If that matches we will return the response. Otherwise we will return null. At this point we will continue to respond to other requests if they match but most likely they wont and at the end we'll throw the error that the cassette was not played entirely.